### PR TITLE
Add can-i checks to linkerd check --pre

### LINKED ISSUE #1472: Add pre-install check to validate that user has permissions to install linkerd
We should add an additional pre-install check (depends on #1470) to validate that the user has the right permissions to install linkerd. Based on the current linkerd install, the user must be able to create

* namespaces
* serviceaccounts
* clusterroles
* clusterrolebindings
* services
* deployments
* configmaps

### PR DESCRIPTION
**Pull request by adleong**
Fixes #1472 

Add checks to `linkerd check --pre` to verify that the user has permission to create:
* namespaces
* serviceaccounts
* clusterroles
* clusterrolebindings
* services
* deployments
* configmaps

Signed-off-by: Alex Leong <alex@buoyant.io>

### CODE REVIEW

**Comment by adleong** [comment]
@dadjeibaah I can definitely see this both ways.  I think I slightly prefer the current implementation because it shows the types of resources that Linkerd will create.

File: pkg/healthcheck/healthcheck.go [hunk 1]
```diff
@@ -100,6 +102,7 @@ type HealthChecker struct {
 	// these fields are set in the process of running checks
 	kubeAPI       *k8s.KubernetesAPI
 	httpClient    *http.Client
+	clientset     *kubernetes.Clientset
```
[the remark this answers is not in the document]
Comment (adleong) [reply pkg/healthcheck/healthcheck.go]: The KubernetesAPI object holds a raw http client that we use to make API calls directly.  Clientset is a higher level API which abstracts us from the API calls.

File: pkg/healthcheck/healthcheck.go [hunk 2]
```diff
@@ -183,6 +186,69 @@ func (hc *HealthChecker) addLinkerdPreInstallChecks() {
 			return nil
 		},
 	})
+
+	hc.checkers = append(hc.checkers, &checker{
+		category:    LinkerdPreInstallCategory,
```
[the remark this answers is not in the document]
Comment (adleong) [reply pkg/healthcheck/healthcheck.go]: maybe we should just change the name of the LinkerdPreInstallCategory.  "kubernetes-setup"? "kubernetes-state"? "kubernetes-permissions"?

[the remark this answers is not in the document]
Comment (rmars) [reply pkg/healthcheck/healthcheck.go]: ooh yeah, I like that. `kubernetes-setup` or `kubernetes-state` sound good to me!

File: pkg/healthcheck/healthcheck.go [hunk 3]
```diff
@@ -183,6 +186,69 @@ func (hc *HealthChecker) addLinkerdPreInstallChecks() {
...
+			return hc.checkCanCreate(hc.ControlPlaneNamespace, "extensions", "v1beta1", "Deployments")
+		},
+	})
+
+	hc.checkers = append(hc.checkers, &checker{
+		category:    LinkerdPreInstallCategory,
+		description: "can create ConfigMaps",
+		fatal:       true,
+		check: func() error {
+			return hc.checkCanCreate(hc.ControlPlaneNamespace, "", "v1", "ConfigMap")
```
[the remark this answers is not in the document]
Comment (klingerf) [reply pkg/healthcheck/healthcheck.go]: This is a good point -- I think we would need to check that the user can create configmaps and secrets in all namespaces if TLS is enabled, but... I'm inclined to leave the checks as you have them in this branch. I'd rather check for the minimal set of permissions, and maybe down the road we could update check to support a `--tls` flag.

**Pull request merged**